In [36]:
import pandas as pd
import matplotlib.pyplot as plt
import datetime as dt
import matplotlib.dates as mdates
import numpy as np
from sklearn.ensemble import RandomForestRegressor
from sklearn.metrics import mean_squared_error
from sklearn.cross_validation import train_test_split

x = pd.read_csv('type_weekend.csv')

for a in range(1,8):
    x['Day_'+str(a)] = 0

for b in range(1,8):
    x['Day_'+str(b)][x['Weekday'] == b] = 1

x.to_csv('type_weekdaycoded.csv')

C:\Users\Chella Rm\Anaconda2\lib\site-packages\ipykernel\__main__.py:16: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy


In [37]:
clf = RandomForestRegressor(n_estimators =100)

### Try removing outliers from Revenue and try

In [38]:
y = pd.read_csv('type_weekdaycoded_formatted.csv')


In [39]:
x_train, x_test, y_train, y_test = train_test_split(y[y.columns[2:22]],y['Revenue(Sales)'],test_size = 0.4)

In [40]:
clf.fit(x_train,y_train)

RandomForestRegressor(bootstrap=True, criterion='mse', max_depth=None,
           max_features='auto', max_leaf_nodes=None, min_samples_leaf=1,
           min_samples_split=2, min_weight_fraction_leaf=0.0,
           n_estimators=100, n_jobs=1, oob_score=False, random_state=None,
           verbose=0, warm_start=False)

In [41]:
print clf.score(x_test,y_test)
y_pred = clf.predict(x_test)

0.955032916854


In [42]:
mean_squared_error(y_test,y_pred)

438847.47195642203

In [43]:
z = pd.read_csv('type_store900.csv')

In [47]:
for la in range(900,1116):
    z['Store_'+str(la)] = 0

In [48]:
z.to_csv('type_store1115.csv')

In [49]:
for la in range(900,1116):
    z['Store_'+str(la)][z['Store']==la] = 1

In [52]:
for la in range(1,100):
    z['Store_'+str(la)][z['Store']==la] = 1

In [53]:
for la in range(100,300):
    z['Store_'+str(la)][z['Store']==la] = 1

In [54]:
for la in range(300,500):
    z['Store_'+str(la)][z['Store']==la] = 1

In [55]:
for la in range(500,700):
    z['Store_'+str(la)][z['Store']==la] = 1

In [56]:
for la in range(700,900):
    z['Store_'+str(la)][z['Store']==la] = 1

In [57]:
for la in range(900,1116):
    z['Store_'+str(la)][z['Store']==la] = 1

In [58]:
z.to_csv('type_store1116.csv')

In [60]:
z['Store_1'][z['Store']==1] = 1

In [61]:
z['Store_1'][z['Store']==1]


0         0
1115      0
2230      0
3345      0
4493      0
5608      0
7155      0
8269      0
9384      0
10531     0
11646     0
12761     0
13876     0
14991     0
16171     0
17286     0
18401     0
19516     0
20631     0
21746     0
22893     0
24008     0
25154     0
26268     0
27382     0
28528     0
29642     0
30756     0
31870     0
32983     0
         ..
259939    0
260872    0
261805    0
262737    0
263669    0
264601    0
265564    0
266496    0
267428    0
268358    0
269289    0
270220    0
271182    0
272113    0
273044    0
273970    0
274897    0
275824    0
276782    0
277709    0
278636    0
279563    0
280490    0
281417    0
282375    0
283303    0
284235    0
285167    0
286099    0
287031    0
Name: Store_1, dtype: int64

In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import datetime as dt
import matplotlib.dates as mdates
import numpy as np
from sklearn.ensemble import RandomForestRegressor
from sklearn.metrics import mean_squared_error
from sklearn.cross_validation import train_test_split

In [2]:
x = pd.read_csv('type_new_1.csv')

In [5]:
x[x['OpenSince']<0]

Unnamed: 0  Store      Date  Weekday  People  Typea  Typec  Typeb  \
130            130    131  11-06-15        4     544      0      1      0   
136            136    137  11-06-15        4     969      1      0      0   
303            303    304  11-06-15        4    1028      1      0      0   
402            402    403  11-06-15        4     739      1      0      0   
687            687    688  11-06-15        4     345      1      0      0   
858            858    859  11-06-15        4     623      0      1      0   
943            943    944  11-06-15        4     908      0      1      0   
995            995    996  11-06-15        4     685      0      1      0   
1019          1019   1020  11-06-15        4     844      1      0      0   
1052          1052   1053  11-06-15        4     879      1      0      0   
1071          1071   1072  11-06-15        4     611      1      0      0   
1245          1245    131  10-06-15        3     493      0      1      0   
1251          1251    137  10-06-15        3     966      1      0      0   
1418          1418    304  10-06-15        3     958      1      0      0   
1517          1517    403  10-06-15        3     716      1      0      0   
1802          1802    688  10-06-15        3     286      1      0      0   
1973          1973    859  10-06-15        3     675      0      1      0   
2058          2058    944  10-06-15        3     978      0      1      0   
2110          2110    996  10-06-15        3     635      0      1      0   
2134          2134   1020  10-06-15        3     933      1      0      0   
2167          2167   1053  10-06-15        3     966      1      0      0   
2186          2186   1072  10-06-15        3     681      1      0      0   
2360          2360    131  09-06-15        2     566      0      1      0   
2366          2366    137  09-06-15        2     964      1      0      0   
2533          2533    304  09-06-15        2    1010      1      0      0   
2632          2632    403  09-06-15        2     764      1      0      0   
2917          2917    688  09-06-15        2     336      1      0      0   
3088          3088    859  09-06-15        2     702      0      1      0   
3173          3173    944  09-06-15        2    1048      0      1      0   
3225          3225    996  09-06-15        2     711      0      1      0   
...            ...    ...       ...      ...     ...    ...    ...    ...   
287530      287530    595  07-07-14        1    1669      0      1      0   
287533      287533    599  07-07-14        1     850      0      0      0   
287558      287558    630  07-07-14        1     823      1      0      0   
287574      287574    653  07-07-14        1     557      0      0      0   
287599      287599    683  07-07-14        1     818      1      0      0   
287603      287603    688  07-07-14        1     308      1      0      0   
287624      287624    718  07-07-14        1     941      1      0      0   
287670      287670    770  07-07-14        1     805      1      0      0   
287681      287681    784  07-07-14        1     887      1      0      0   
287696      287696    800  07-07-14        1     512      0      0      0   
287711      287711    819  07-07-14        1     575      1      0      0   
287718      287718    828  07-07-14        1     614      0      0      0   
287727      287727    837  07-07-14        1     557      1      0      0   
287729      287729    839  07-07-14        1     730      0      1      0   
287758      287758    872  07-07-14        1     812      1      0      0   
287764      287764    878  07-07-14        1     878      0      0      0   
287794      287794    918  07-07-14        1     553      1      0      0   
287816      287816    944  07-07-14        1    1053      0      1      0   
287827      287827    956  07-07-14        1     561      1      0      0   
287841      287841    970  07-07-14        1     481      1      0      0   
2878

In [6]:
x['Store_1'][0] 


0

In [3]:
for i in range(0,len(x)):
    x['Store_'+str(x['Store'][i])][i] = 1

In [6]:
y = pd.read_csv('DOS.csv')

## Remaining Work:
### Promo2Since can possibly be coded as year value instead of nth promotion
### The 1115 column data set could not be manipulated
### Can apply a log transform to the competitiondistance 
### Have to preprocess the test data (Very important)
### Look for other transforms
### Fine tune the tree if possible

### log(1/x)
### What to do about outliers in the sales? lets check if its from a certain shop or scarce

In [7]:
t = int(y['CompetitionOpenSinceYear'].mean())
u = int(y['CompetitionOpenSinceMonth'].mean())
y['CompetitionOpenSinceYear'][y['CompetitionOpenSinceYear'].isnull()] = t
y['CompetitionOpenSinceMonth'][y['CompetitionOpenSinceMonth'].isnull()] = u

C:\Users\Chella Rm\Anaconda2\lib\site-packages\ipykernel\__main__.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  app.launch_new_instance()
C:\Users\Chella Rm\Anaconda2\lib\site-packages\ipykernel\__main__.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy


In [8]:
y['CompetitionOpenSince'] = y['CompetitionOpenSinceYear'] + y['CompetitionOpenSinceMonth']/12


In [10]:
y.to_csv('DISnew.csv')

In [13]:
y['Promo2Date'] = 0.0
for g in range(len(y)):
    if y['Promo2'][g]==1:
        y['Promo2Date'][g] = y['Promo2SinceYear'][g]+ (y['Promo2SinceWeek'][g]/52)
        
        

C:\Users\Chella Rm\Anaconda2\lib\site-packages\ipykernel\__main__.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy


In [14]:
y.to_csv('DISSY.csv')

In [16]:
y['CompetitionDistance'].describe()

count     1112.000000
mean      5404.901079
std       7663.174720
min         20.000000
25%        717.500000
50%       2325.000000
75%       6882.500000
max      75860.000000
Name: CompetitionDistance, dtype: float64